.github/changelog.py

```diff
@@ -0,0 +1,20 @@
+import sys
+import re
+
+def main():
+    version = sys.argv[1]
+
+    # Recognise headers of the form ### 14.0 (2024-04-01) and ## 14.x but not ### 15.0-beta1 (2024-12-24)
+    stable_header = re.compile(r"^#* ([0-9]+.[0-9x]+)([^-]|$)")
+
+    with open("changelog.md", "r") as file:
+        # We never want to output an empty changelog, even if a header is on the first line.
+        print(next(file), end="")
+        for line in file:
+            m = re.match(stable_header, line)
+            if m is not None and m.group(1) != version:
+                break
+            print(line, end="")
```

Comment:
Commonly we use `sys.stdout.write(line)` if we want to keep the (CR)LF of the original, or we just use `print(line.strip())`, letting Python output the right (CR)LF. But as the file opened with `"r"`, newlines are being processed by Python anyway, so it is all a bit weird.

I personally flavour the `strip()` approach, just to be clear: what-ever Python things is right :P